chore_: network providers reorder

saledjenic opened the pull request:
A new order of using providers is:
1. StatusProxy - Node Fleet
2. StatusProxy - Infura
3. Direct Infura
4. StatusProxy - Grove
5. Direct Grove

Closes #5944

status-im-auto commented:
### Jenkins Builds
<details>
<summary>Click to see older builds (47)</summary>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 8c4ba203 | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5941/1/) | 2024-10-14 09:20:27 | ~5 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-241014-091506-8c4ba2-pr5941.zip) |
| :heavy_check_mark: | 8c4ba203 | [#1](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5941/1/) | 2024-10-14 09:20:27 | ~5 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5941/1//consoleText) |
| :heavy_check_mark: | 8c4ba203 | [#1](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5941/1/) | 2024-10-14 09:20:41 | ~5 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-241014-091502-8c4ba2-pr5941.zip) |
| :heavy_check_mark: | 8c4ba203 | [#1](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5941/1/) | 2024-10-14 09:21:03 | ~6 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-241014-091501-8c4ba2-pr5941.aar) |
| :heavy_check_mark: | 8c4ba203 | [#1](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5941/1/) | 2024-10-14 09:49:16 | ~34 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5941/1//consoleText) |
| | | | | | | |
| :heavy_check_mark: | e9c61734 | [#2](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5941/2/) | 2024-10-14 14:42:45 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-241014-144009-e9c617-pr5941.aar) |
| :heavy_check_mark: | e9c61734 | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5941/2/) | 2024-10-14 14:43:09 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-241014-144013-e9c617-pr5941.zip) |
| :heavy_check_mark: | e9c61734 | [#2](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5941/2/) | 2024-10-14 14:44:15 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-241014-144010-e9c617-pr5941.zip) |
| :heavy_check_mark: | e9c61734 | [#2](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5941/2/) | 2024-10-14 14:45:23 | ~5 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5941/2//consoleText) |
| :heavy_multiplication_x: | e9c61734 | [#2](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5941/2/) | 2024-10-14 15:12:29 | ~32 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5941/2//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 937aada5 | [#3](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5941/3/) | 2024-10-14 19:50:54 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-241014-194819-937aad-pr5941.aar) |
| :heavy_check_mark: | 937aada5 | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5941/3/) | 2024-10-14 19:51:21 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-241014-194824-937aad-pr5941.zip) |
| :heavy_check_mark: | 937aada5 | [#3](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5941/3/) | 2024-10-14 19:52:11 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-241014-194820-937aad-pr5941.zip) |
| :heavy_multiplication_x: | 937aada5 | [#3](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5941/3/) | 2024-10-14 19:53:41 | ~5 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5941/3//consoleText) |
| :heavy_multiplication_x: | 937aada5 | [#3](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5941/3/) | 2024-10-14 20:21:36 | ~33 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5941/3//consoleText) |
| :heavy_multiplication_x: | 937aada5 | [#4](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5941/4/) | 2024-10-15 06:25:52 | ~1 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5941/4//consoleText) |
| :heavy_multiplication_x: | 937aada5 | [#4](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5941/4/) | 2024-10-15 06:57:05 | ~32 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5941/4//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 53db1bb9 | [#4](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5941/4/) | 2024-10-15 08:02:35 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-241015-080003-53db1b-pr5941.aar) |
| :heavy_check_mark: | 53db1bb9 | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5941/4/) | 2024-10-15 08:03:05 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-241015-080004-53db1b-pr5941.zip) |
| :heavy_check_mark: | 53db1bb9 | [#4](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5941/4/) | 2024-10-15 08:04:19 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-241015-080003-53db1b-pr5941.zip) |
| :heavy_multiplication_x: | 53db1bb9 | [#5](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5941/5/) | 2024-10-15 08:05:14 | ~5 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5941/5//consoleText) |
| :heavy_multiplication_x: | 53db1bb9 | [#5](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5941/5/) | 2024-10-15 08:33:25 | ~33 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5941/5//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 66c7bb86 | [#5](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5941/5/) | 2024-10-15 08:51:15 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-241015-084927-66c7bb-pr5941.aar) |
| :heavy_check_mark: | 66c7bb86 | [#5](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5941/5/) | 2024-10-15 08:51:59 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-241015-084929-66c7bb-pr5941.zip) |
| :heavy_check_mark: | 66c7bb86 | [#5](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5941/5/) | 2024-10-15 08:52:42 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-241015-084928-66c7bb-pr5941.zip) |
| :heavy_multiplication_x: | 66c7bb86 | [#6](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5941/6/) | 2024-10-15 08:54:24 | ~4 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5941/6//consoleText) |
| :heavy_check_mark: | 66c7bb86 | [#6](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5941/6/) | 2024-10-15 09:22:06 | ~32 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5941/6//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 4add067e | [#6](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5941/6/) | 2024-10-15 09:43:19 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-241015-094052-4add06-pr5941.aar) |
| :heavy_check_mark: | 4add067e | [#6](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5941/6/) | 2024-10-15 09:43:40 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-241015-094054-4add06-pr5941.zip) |
| :heavy_check_mark: | 4add067e | [#6](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5941/6/) | 2024-10-15 09:45:02 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-241015-094053-4add06-pr5941.zip) |
| :heavy_check_mark: | 4add067e | [#7](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5941/7/) | 2024-10-15 09:46:00 | ~5 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5941/7//consoleText) |
| :heavy_check_mark: | 4add067e | [#7](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5941/7/) | 2024-10-15 10:13:07 | ~32 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5941/7//consoleText) |
| | | | | | | |
| :heavy_multiplication_x: | da4d225d | [#8](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5941/8/) | 2024-10-15 13:00:44 | ~1 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5941/8//consoleText) |
| :heavy_check_mark: | da4d225d | [#7](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5941/7/) | 2024-10-15 13:01:29 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-241015-125903-da4d22-pr5941.aar) |
| :heavy_check_mark: | da4d225d | [#7](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5941/7/) | 2024-10-15 13:02:03 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-241015-125908-da4d22-pr5941.zip) |
| :heavy_check_mark: | da4d225d | [#7](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5941/7/) | 2024-10-15 13:02:54 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-241015-125904-da4d22-pr5941.zip) |
| :heavy_check_mark: | da4d225d | [#8](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5941/8/) | 2024-10-15 13:04:30 | ~5 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5941/8//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 2226a8a9 | [#8](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5941/8/) | 2024-10-15 13:03:31 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-241015-130143-2226a8-pr5941.aar) |
| :heavy_check_mark: | 2226a8a9 | [#8](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5941/8/) | 2024-10-15 13:05:01 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-241015-130217-2226a8-pr5941.zip) |
| :heavy_check_mark: | 2226a8a9 | [#8](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5941/8/) | 2024-10-15 13:07:05 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-241015-130308-2226a8-pr5941.zip) |
| :heavy_check_mark: | 2226a8a9 | [#9](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5941/9/) | 2024-10-15 13:10:15 | ~5 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5941/9//consoleText) |
| :heavy_check_mark: | 2226a8a9 | [#9](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5941/9/) | 2024-10-15 13:32:56 | ~32 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5941/9//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 6979a9b0 | [#9](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5941/9/) | 2024-10-15 13:13:04 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-241015-131030-6979a9-pr5941.aar) |
| :heavy_check_mark: | 6979a9b0 | [#9](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5941/9/) | 2024-10-15 13:14:22 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-241015-131031-6979a9-pr5941.zip) |
| :heavy_check_mark: | 6979a9b0 | [#9](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5941/9/) | 2024-10-15 13:15:03 | ~4 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-241015-131034-6979a9-pr5941.zip) |
| :heavy_check_mark: | 6979a9b0 | [#10](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5941/10/) | 2024-10-15 13:15:55 | ~5 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5941/10//consoleText) |
| :heavy_check_mark: | 6979a9b0 | [#10](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5941/10/) | 2024-10-15 14:05:56 | ~32 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5941/10//consoleText) |
</details>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 93e88d97 | [#10](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5941/10/) | 2024-10-16 09:48:52 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-241016-094614-93e88d-pr5941.aar) |
| :heavy_check_mark: | 93e88d97 | [#10](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5941/10/) | 2024-10-16 09:49:27 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-241016-094618-93e88d-pr5941.zip) |
| :heavy_check_mark: | 93e88d97 | [#10](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5941/10/) | 2024-10-16 09:50:32 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-241016-094615-93e88d-pr5941.zip) |
| :heavy_check_mark: | 93e88d97 | [#11](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5941/11/) | 2024-10-16 09:51:39 | ~5 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5941/11//consoleText) |
| :heavy_check_mark: | 93e88d97 | [#11](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5941/11/) | 2024-10-16 10:19:08 | ~32 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5941/11//consoleText) |
| | | | | | | |
| :x: | b0375be2 | [#11](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-5941/11/) | 2024-10-21 13:02:23 | ~12 sec | `linux` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-5941/11/consoleText) |
| :heavy_check_mark: | b0375be2 | [#11](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5941/11/) | 2024-10-21 13:06:48 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-241021-130206-b0375b-pr5941.zip) |
| :heavy_check_mark: | b0375be2 | [#11](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5941/11/) | 2024-10-21 13:07:06 | ~5 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-241021-130205-b0375b-pr5941.aar) |
| :heavy_check_mark: | b0375be2 | [#12](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5941/12/) | 2024-10-21 13:07:26 | ~5 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5941/12//consoleText) |
| :x: | b0375be2 | [#12](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-5941/12/) | 2024-10-21 13:09:02 | ~11 sec | `linux` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-5941/12/consoleText) |
| :x: | b0375be2 | [#13](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-5941/13/) | 2024-10-21 13:18:30 | ~11 sec | `linux` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-5941/13/consoleText) |
| :x: | b0375be2 | [#14](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-5941/14/) | 2024-10-21 13:26:25 | ~11 sec | `linux` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-5941/14/consoleText) |
| :x: | b0375be2 | [#15](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-5941/15/) | 2024-10-21 13:35:09 | ~13 sec | `linux` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-5941/15/consoleText) |
| :heavy_check_mark: | b0375be2 | [#12](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5941/12/) | 2024-10-21 13:35:47 | ~33 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5941/12//consoleText) |
| :x: | b0375be2 | [#16](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-5941/16/) | 2024-10-21 13:37:42 | ~10 sec | `linux` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-5941/16/consoleText) |
| :x: | b0375be2 | [#17](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-5941/17/) | 2024-10-21 13:39:55 | ~10 sec | `linux` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-5941/17/consoleText) |
| :x: | b0375be2 | [#18](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-5941/18/) | 2024-10-21 13:48:28 | ~8.6 sec | `linux` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-5941/18/consoleText) |
| :x: | b0375be2 | [#19](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-5941/19/) | 2024-10-21 13:49:40 | ~11 sec | `linux` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-5941/19/consoleText) |
| :x: | b0375be2 | [#20](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-5941/20/) | 2024-10-21 13:58:34 | ~10 sec | `linux` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-5941/20/consoleText) |
| :x: | b0375be2 | [#21](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-5941/21/) | 2024-10-21 13:59:21 | ~11 sec | `linux` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-5941/21/consoleText) |
| :x: | b0375be2 | [#22](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-5941/22/) | 2024-10-21 14:30:31 | ~11 sec | `linux` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-5941/22/consoleText) |
| :heavy_check_mark: | b0375be2 | [#23](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-5941/23/) | 2024-10-21 15:30:26 | ~4 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-241021-152600-b0375b-pr5941.zip) |

shivekkhurana commented:
We should keep Grove because it's the only decentralised provider we have. And it doesn't cost us anything to run.

We can change the order of fallback so the requests don't go there, but if there is a state banning event on Infura (or our proxy), Grove can potentially keep the app usable.

saledjenic commented:
@shivekkhurana I am not sure why we would like to keep Grove, as an option, if we proved multiple times that it's unreliable, not only that it returns errors where it should not, but also that it returns inaccurate responses and that's more dangerous.

saledjenic committed:
chore_: providers re-ordered

A new order of using providers is:
1. StatusProxy - Node Fleet
2. StatusProxy - Infura
3. Direct Infura
4. StatusProxy - Grove
5. Direct Grove

saledjenic commented:
@status-im/status-go-guild what do you think of updating `Networks` type and instead of having properties `DefaultRPCURL`, `DefaultFallbackURL`, `DefaultFallbackURL2`, `RPCURL`, `FallbackURL` is better to have an array of rpc URL and use that order?

shivekkhurana commented:
> @shivekkhurana I am not sure why we would like to keep Grove, as an option, if we proved multiple times that it's unreliable, not only that it returns errors where it should not, but also that it returns inaccurate responses and that's more dangerous.

As discussed on the call. Grove is the most low prio RPC for doomsday scenarios.

friofry commented on api/default_networks.go:
```diff
@@ -31,9 +31,9 @@ func mainnet(stageName string) params.Network {
-		DefaultFallbackURL2:    fmt.Sprintf("https://%s.api.status.im/grove/ethereum/mainnet/", stageName),
-		RPCURL:                 "https://eth-archival.rpc.grove.city/v1/",
-		FallbackURL:            "https://mainnet.infura.io/v3/",
+		DefaultFallbackURL2:    "https://mainnet.infura.io/v3/",
```
```
	if proxyProvider.Enabled {
		key := ProviderStatusProxy
		keyFallback := ProviderStatusProxy + "-fallback"
		keyFallback2 := ProviderStatusProxy + "-fallback2"
		urls[key] = network.DefaultRPCURL
		urls[keyFallback] = network.DefaultFallbackURL
		urls[keyFallback2] = network.DefaultFallbackURL2
		keys = []string{key, keyFallback, keyFallback2}
		authMap[key] = proxyProvider.User + ":" + proxyProvider.Password
		authMap[keyFallback] = authMap[key]
		authMap[keyFallback2] = authMap[key]
	}
	keys = append(keys, []string{"main", "fallback"}...)
	urls["main"] = network.RPCURL
	urls["fallback"] = network.FallbackURL
```
We do not use authentication for RPCURL. If I understand correctly it should be:
```
		DefaultRPCURL:          fmt.Sprintf("https://%s.api.status.im/nodefleet/ethereum/mainnet/", stageName),
		DefaultFallbackURL:     fmt.Sprintf("https://%s.api.status.im/infura/ethereum/mainnet/", stageName),
		DefaultFallbackURL2:    fmt.Sprintf("https://%s.api.status.im/grove/ethereum/mainnet/", stageName),
		RPCURL:                 "https://mainnet.infura.io/v3/",
		FallbackURL:            "https://eth-archival.rpc.grove.city/v1/",
		BlockExplorerURL:       "https://etherscan.io/",
```
```
StatusProxy - Node Fleet
StatusProxy - Infura
Direct Infura
StatusProxy - Grove
Direct Grove
```
and if we need to have `proxy-grove` lower than `direct-infura`, then we should probably add "priority" field to params.network, and respect it in `rpc/client.go` `getEthClients` method.

saledjenic replied:
@friofry good point, I missed that usage. Thanks.
Need to think a bit about how to do it painless for now.

saledjenic replied:
Now should be good, providers priority is determined by the `getProviderPriorityByURL` function.

friofry commented:
> @status-im/status-go-guild what do you think of updating `Networks` type and instead of having properties `DefaultRPCURL`, `DefaultFallbackURL`, `DefaultFallbackURL2`, `RPCURL`, `FallbackURL` is better to have an array of rpc URL and use that order?

I think it would be great to introduce a one-to-many relationship between `Network` and `Provider`. But I would suggest to make it out of the scope of this task

dlipicar commented on api/default_networks_test.go:
```diff
@@ -25,68 +25,42 @@ func TestBuildDefaultNetworks(t *testing.T) {
-	require.True(t, strings.Contains(actualNetworks[8].DefaultRPCURL, stageName))
-	require.True(t, strings.Contains(actualNetworks[8].DefaultFallbackURL, stageName))
-	require.True(t, strings.Contains(actualNetworks[8].DefaultFallbackURL2, stageName))
+	ignoreDefaultRPCURLCheck := false // TODO: used just because of Goerli, remove once we remove Goerli from the default networks
```
It's fine to leave it if removing Goerli now will introduce too much complexity in this PR, there's 0 reason to have it since none of the providers we use support it still.

saledjenic replied:
Here is a new PR that removes Goerli from the code https://github.com/status-im/status-go/pull/5945

saledjenic committed:
chore_: completely remove goerli from the code

friofry commented on api/default_networks.go:
```diff
@@ -251,25 +251,40 @@ func setRPCs(networks []params.Network, request *requests.WalletSecretsConfig) [
+		grove  = "grove.city/"
+	)
+
 	for _, n := range networks {
```
My suggestion is to add a functor to make it a bit shorter and avoid duplication:
```
func setRPCs(networks []params.Network, request *requests.WalletSecretsConfig) []params.Network {

	var networksWithRPC []params.Network

	const (
		infura = "infura.io/"
		grove  = "grove.city/"
	)

	appendToken := func(url string) string {
		if strings.Contains(url, infura) && request.InfuraToken != "" {
			return url + request.InfuraToken
		} else if strings.Contains(url, grove) && request.PoktToken != "" {
			return url + request.PoktToken
		}
		return url
	}

	for _, n := range networks {
		n.DefaultRPCURL = appendToken(n.DefaultRPCURL)
		n.DefaultFallbackURL = appendToken(n.DefaultFallbackURL)
		n.DefaultFallbackURL2 = appendToken(n.DefaultFallbackURL2)
		n.RPCURL = appendToken(n.RPCURL)
		n.FallbackURL = appendToken(n.FallbackURL)

		if request.GanacheURL != "" {
			n.RPCURL = request.GanacheURL
			n.FallbackURL = request.GanacheURL
			switch n.ChainID {
			case mainnetChainID:
				n.TokenOverrides = []params.TokenOverride{mainnetGanacheTokenOverrides}
			case goerliChainID:
				n.TokenOverrides = []params.TokenOverride{goerliGanacheTokenOverrides}
			}
		}

		networksWithRPC = append(networksWithRPC, n)
	}

	return networksWithRPC
}

```

friofry commented on api/default_networks.go:
```diff
@@ -75,8 +75,8 @@ func sepolia(stageName string) params.Network {
 		DefaultFallbackURL2:    fmt.Sprintf("https://%s.api.status.im/grove/ethereum/sepolia/", stageName),
-		RPCURL:                 "https://sepolia-archival.rpc.grove.city/v1/",
-		FallbackURL:            "https://sepolia.infura.io/v3/",
+		RPCURL:                 "https://sepolia.infura.io/v3/",
```
if proxy can be in non-`DefaultUrl`, maybe rename all of them to RpcUrl1, RpcUrl2, RpcUrl3, RpcUrl4, RpcUrl5 ?

friofry commented on rpc/client.go:
```diff
@@ -213,9 +213,32 @@ func (c *Client) getClientUsingCache(chainID uint64) (chain.ClientInterface, err
+}
+
 func (c *Client) getEthClients(network *params.Network) []ethclient.RPSLimitedEthClientInterface {
+	// Keys are defining the order of providers being used
```
I see 2 problems:
- if the same key priority is used 
- some code duplication

my suggestion to add something like this:

```
type Provider struct {
	Key      string
	URL      string
	Auth     string
	Priority int
}

func createProvider(key, url, credentials string, providers *[]Provider) {
	priority := getProviderPriorityByURL(url)
	*providers = append(*providers, Provider{
		Key:      key,
		URL:      url,
		Auth:     credentials,
		Priority: priority,
	})
}

func (c *Client) prepareProviders(network *params.Network) []Provider {
	var providers []Provider

	// Retrieve the proxy provider configuration
	proxyProvider, err := getProviderConfig(c.providerConfigs, ProviderStatusProxy)
	if err != nil {
		c.log.Warn("could not find provider config for status-proxy", "error", err)
	}

	// Add main and fallback providers
	createProvider(ProviderMain, network.RPCURL, "", &providers)
	createProvider(ProviderFallback, network.FallbackURL, "", &providers)

	// If the proxy provider is enabled, add it and its fallback options
	if proxyProvider.Enabled {
		credentials := proxyProvider.User + ":" + proxyProvider.Password
		createProvider(ProviderStatusProxy, network.DefaultRPCURL, credentials, &providers)
		createProvider(ProviderStatusProxyFallback, network.DefaultFallbackURL, credentials, &providers)
		createProvider(ProviderStatusProxyFallback2, network.DefaultFallbackURL2, credentials, &providers)
	}

	// Sort providers by priority
	sort.Slice(providers, func(i, j int) bool {
		return providers[i].Priority < providers[j].Priority
	})

	return providers
}
```

and use it in `getEthClients`

saledjenic commented:
@friofry thanks for the suggestions, I agree it's better, but I wanted to do it with minimal changes to the current code, that's why I introduced `getProviderPriorityByURL`. Also as mentioned above leave all those optimizations and improvements for the moment when we're going to remove props `DefaultRPCURL`, `DefaultFallbackURL`, `DefaultFallbackURL2`... and switch to array or map of providers with all necessary fields and proper validations, including priority property.

saledjenic commented:
> if the same key priority is used

That's possible if the same URLs are used for different networks, but that check should be part of the config validation process.

saledjenic commented:
@friofry updated the code as you suggested. Thanks.

friofry commented:
> ee it's better, but I wanted to do it with minimal changes to the current code, that's why I introduced `getProviderPriorityByURL`. Also as mentioned above leave al

Yeah, I realised that too and deleted my comment :)

friofry commented on rpc/provider.go:
```diff
@@ -0,0 +1,96 @@
+	ProviderFallback             = "fallback"
+	ProviderStatusProxy          = "status-proxy"
+	ProviderStatusProxyFallback  = ProviderStatusProxy + "-fallback"
+	ProviderStatusProxyFallback2 = ProviderStatusProxy + "-fallback"
```
I guess it's a typo, should be`fallback2`

saledjenic replied:
You were faster in commenting on that than me updating it. :)
Thanks, updated.

friofry reviewed (approved):
Cool!
